File: app/src/main/java/com/left/gank/ui/MainActivity.java
package com.left.gank.ui;

import android.os.Bundle;

import com.left.gank.R;
import com.left.gank.rxjava.RxBus_;
import com.left.gank.ui.base.activity.BaseActivity;
import com.left.gank.ui.discovered.DiscoveredFragment;
import com.left.gank.ui.girls.GirlsFragment;
import com.left.gank.ui.index.IndexFragment;
import com.left.gank.ui.mine.MineFragment;
import com.roughike.bottombar.BottomBar;
import com.roughike.bottombar.OnTabSelectListener;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import butterknife.BindView;

/**
 * Create by LingYan on 2016-6-13
 */
public class MainActivity extends BaseActivity {
    private static final int DEFAULT_TAB_POSITION = 0;

    private static final int TAB_HOME = 0;
    private static final int TAB_NEWS = 1;
    private static final int TAB_IMAGE = 2;
    private static final int TAB_MORE = 3;

    @BindView(R.id.bottomBar)
    BottomBar bottomBar;

    private Fragment curFragment;
    private List<Fragment> fragmentList;

    @Override
    protected int getContentId() {
        return R.layout.fragment_main_bottom_navigation;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        fragmentList = getFragmentList();
        bottomBar.setDefaultTabPosition(DEFAULT_TAB_POSITION);
        bottomBar.setOnTabSelectListener(onTabSelectListener);
    }

    private final OnTabSelectListener onTabSelectListener = tabId -> {
        int index = getFragmentIndex(tabId);
        openFragment(index);
    };

    private int getFragmentIndex(int tabId) {
        int index = TAB_HOME;
        switch (tabId) {
            case R.id.tab_home:
                index = TAB_HOME;
                break;
            case R.id.tab_news:
                index = TAB_NEWS;
                break;
            case R.id.tab_image:
                index = TAB_IMAGE;
                break;
            case R.id.tab_more:
                index = TAB_MORE;
                break;
        }
        return index;
    }

    private void openFragment(int index) {
        Fragment fragmentTo = fragmentList.get(index);
        if (curFragment == null) {
            addMainFragment(fragmentTo);
            curFragment = fragmentTo;
        } else {
            if (!curFragment.equals(fragmentTo)) {
                addAnimFragment(curFragment, fragmentTo, true);
                curFragment = fragmentTo;
            }
        }
    }

    private List<Fragment> getFragmentList() {
        List<Fragment> fragments = new ArrayList<>();
        fragments.add(new IndexFragment());
        fragments.add(new DiscoveredFragment());
        fragments.add(new GirlsFragment());
        fragments.add(new MineFragment());
        return fragments;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        RxBus_.getInstance().removeAllStickyEvents();// 移除所有Sticky事件
    }
}

File: app/src/main/java/com/left/gank/ui/MainActivity.kt
package com.left.gank.ui

import android.os.Bundle
import android.rxbus.RxApiManager
import androidx.fragment.app.Fragment
import com.left.gank.R
import com.left.gank.base.activity.BaseActivity
import com.left.gank.ui.discovered.DiscoveredFragment
import com.left.gank.ui.girls.GirlsFragment
import com.left.gank.ui.index.IndexFragment
import com.left.gank.ui.mine.MineFragment
import kotlinx.android.synthetic.main.fragment_main_bottom_navigation.*

/**
 *
 * Create by LingYan on 2019-06-29
 */
class MainActivity : BaseActivity() {
    private var curFragment: Fragment? = null

    private val fragmentList = listOf(
            IndexFragment(),
            DiscoveredFragment(),
            GirlsFragment(),
            MineFragment()
    )

    companion object {
        const val TAB_HOME = 0
        const val TAB_NEWS = 1
        const val TAB_IMAGE = 2
        const val TAB_MORE = 3
    }

    override fun getContentId(): Int = R.layout.fragment_main_bottom_navigation

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bottom_navigation!!.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.tab_home -> {
                    openFragment(TAB_HOME)
                    true
                }
                R.id.tab_news -> {
                    openFragment(TAB_NEWS)
                    true
                }
                R.id.tab_image -> {
                    openFragment(TAB_IMAGE)
                    true
                }
                R.id.tab_more -> {
                    openFragment(TAB_MORE)
                    true
                }
                else -> false
            }
        }

        bottom_navigation.selectedItemId = R.id.tab_home
    }

    private fun openFragment(index: Int) {
        val toFragment = fragmentList[index]
        if (curFragment == null) {
            addMainFragment(toFragment)
            curFragment = toFragment
        } else {
            if (curFragment != toFragment) {
                addAnimFragment(curFragment, toFragment, true)
                curFragment = toFragment
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        RxApiManager.get().clear()
    }
}